tests/chainer_tests/functions_tests/array_tests/test_split_axis.py

```diff
@@ -138,12 +146,30 @@ def inject_backend_tests():
 @inject_backend_tests()
 class TestSplitAxis(testing.FunctionTestCase):
 
+    grad_outputs_is_none = None
+
     def generate_inputs(self):
         shape = self.shape
         dtype = self.dtype
         x = numpy.arange(numpy.prod(shape), dtype=dtype).reshape(shape)
         return x,
 
+    def _generate_grad_outputs(self, outputs_template):
+        grad_outputs = self.generate_grad_outputs(outputs_template)
+        return grad_outputs
+
+    def generate_grad_outputs(self, outputs_template):
+        grad_outputs = tuple([
+            numpy.random.uniform(-1, 1, a.shape).astype(a.dtype)
+            for a in outputs_template])
+
+        if self.grad_outputs_is_none is not None:
+            grad_outputs = tuple(
+                None if is_none else g for is_none, g,
+                in six.moves.zip(self.grad_outputs_is_none, grad_outputs))
+
+        return grad_outputs
+
```

Comment:
Instead of overriding _generate_grad_outputs, which is "private" and is never overrided in any part of the code.
How about creating an initializer function for a single grad_output and use it in `testing.function_link`.?

Reply:
> overriding `_generate_grad_outputs`

IMO, `forward` (and `forward_expected`) should be fixed instead.

Reply:
Oh, do you mean to make forward to explicitly return `None`?

Reply:
Sorry I should've mentioned that the private method should not be overridden. This PR should be rebased after https://github.com/chainer/chainer/pull/7831.

Reply:
Updated the PR now that #7831 is merged.